Firestore/Source/API/FIRDocumentSnapshot.mm

```diff
@@ -78,10 +79,15 @@ + (instancetype)snapshotWithFirestore:(FIRFirestore *)firestore
                              document:(nullable FSTDocument *)document
                             fromCache:(BOOL)fromCache
                      hasPendingWrites:(BOOL)pendingWrites {
-  DocumentSnapshot underlyingSnapshot{firestore, documentKey, document,
+  DocumentSnapshot underlyingSnapshot{firestore.underlyingFirestore, documentKey, document,
```

Comment:
This is kinda ugly.

Reply:
Yeah. I really didn't like this either. See #2507 for my proposal for how to make this better.